src/AWSCredentials.jl

```diff
@@ -53,30 +50,19 @@ mutable struct AWSCredentials
     user_arn::String
     account_number::String
     expiry::DateTime
+    renew::Base.Callable
 
     function AWSCredentials(access_key_id,secret_key,
                             token="", user_arn="", account_number="";
-                            expiry=now(UTC) + DEFAULT_DURATION)
-        new(access_key_id, secret_key, token, user_arn, account_number, expiry)
+                            expiry=typemax(DateTime),
+                            renew=Nothing)
+        new(access_key_id, secret_key, token, user_arn, account_number, expiry, renew)
     end
 end
 
-"""
-Holds an [`AWSCredentials`](@ref) object and ensures it is renewed before it can expire.
-Use [`get_credentials`](@ref) to access the credentials.
-
-The `RenewableAWSCredentials()` constructor tries to load local Credentials from
-environment variables, `~/.aws/credentials`, `~/.aws/config` or EC2 instance credentials.
-To specify the profile to use from `~/.aws/credentials`, do, for example, `RenewableAWSCredentials(profile="profile-name")`.
-"""
-struct RenewableAWSCredentials
-    creds::AWSCredentials
-    renew::Function
-end
-
-function RenewableAWSCredentials(;profile=nothing)
+function AWSCredentials(;profile=nothing)
     creds = nothing
-    renew = nothing
+    renew = Nothing
```

Comment:
`Nothing` is technically a `Base.Callable`, since it's a `Type`, but it's not really meaningful. If it were me, I'd use `Union{Base.Callable, Nothing}` and set `renew = nothing` here. What do you think about that?

Reply:
I mostly set it to `Nothing` because a couple of the credential getter functions return `nothing` if they fail to find credentials.
The way it stands now, if `renew()` will return nothing if `renew` is set to one of those functions and fails or if `AWSCredentials` isn't supplied with a renewal function.

Reply:
Oh, I see. Okay, thanks for explaining!

Reply:
I can see how it wouldn't be readily apparent. I really should throw in some comments as to the expected behaviour for renewal functions and the like.

I'm still working on documenting my code better, so thanks for catching it!

Reply:
On second thought the aforementioned functions failing should probably be treated as an error since they both check credential files that shouldn't disappear. So your suggestion is probably the best way to go.